gradle/libs.versions.toml

```diff
@@ -1,30 +1,36 @@
+# Version matrix for Gradle, Kotlin, and AGP:
+# https://kotlinlang.org/docs/gradle-configure-project.html#apply-the-plugin
+
+# Version matrix for Compose:
+# https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-compatibility-and-versioning.html#jetpack-compose-artifacts-used
+
 [versions]
-androidx-activity = "1.9.3"
-androidx-composeUi = "1.7.6"
-androidx-navigation = "2.9.0-alpha15"
+androidx-activity = "1.10.1"
+androidx-composeUi = "1.8.1"
+androidx-navigation = "2.9.0-beta01"
 kermit = "2.0.5"
 kotlinx-browser = "0.3"
 kotlinx-coroutines = "1.10.2"
 ktor = "3.1.3"
 maplibre-android-sdk = "11.8.7"
 maplibre-android-plugins = "3.0.2"
-maplibre-ios = "6.9.0"
-maplibre-js = "4.7.1"
-spatialk = "0.3.0"
+maplibre-ios = "6.12.3" # TODO: is outdated; 6.13 isn't available on cocoapods
+maplibre-js = "4.7.1" # TODO: is outdated; update to v5.x
+spatialk = "0.3.0" # TODO: is unmaintained; watch for https://github.com/maplibre/maplibre-java/pull/40
 webpack-html = "5.6.3"
 webpack-htmlInlineScript = "3.2.1"
 webview = "1.9.40"
 
 gradle-android = "8.7.3"
-gradle-compose = "1.7.3"
+gradle-compose = "1.8.0"
 gradle-dokka = "2.0.0"
 gradle-jgitver = "0.10.0-rc03"
-gradle-kotlin = "2.1.20"
-gradle-mavenPublish = "0.30.0"
+gradle-kotlin = "2.1.21-RC2" # TODO is prerelease; move to stable
```

Comment:
Before merging, I should downgrade my local xcode and revert this change (or update it if .21 comes out in the meantime)